### ISSUE TITLE
uctypes does not support values larger than 31 bits despite listing uint64 support

### ISSUE THREAD

**Issue by ryannathans** [micropython/micropython#2269]
```
>>> struct = uctypes.struct(uctypes.addressof(buffer), {"bug": uctypes.UINT32 | 0, "bug64": uctypes.UINT64 | 4})
>>> struct.bug = 2147483647
>>> struct.bug = 2147483648
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: overflow converting long int to machine word
>>> struct.bug64 = 2147483647
>>> struct.bug64 = 2147483648
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: overflow converting long int to machine word
```

Seems pretty funny how ctypes can't support UINT32 and UINT64s?

**Comment by pfalcon** [comment]
To fix this, we need getters for getting Python int object as: int32, uint32, int64, uint64, each apparently "truncated" and "range checking" version. That's many getters, so apparently there should be one new function, taking flags for the above, deprecating previously available mp_obj_int_get_truncated() & mp_obj_int_get_checked().

**Comment by dpgeorge** [comment]
Just using mp_obj_int_get_truncated should fix the 32 bit case.  Then there are already some functions to handle wider values, eg int_to_bytes() and mp_obj_int_to_bytes_impl().

**Comment by pfalcon** [comment]
Ok, I have fixes for these working on 32-bit little-endian, now the most boring part is writing tests...

**Comment by pfalcon** [comment]
Fixed by 9e8f3163924c1f429f16f44a4a27b0cd33064719
